src/simulator/controllers/map/camera_controller.py

```diff
@@ -0,0 +1,184 @@
+from direct.gui.OnscreenText import OnscreenText
+from direct.showbase.ShowBaseGlobal import globalClock
+from direct.showbase.DirectObject import DirectObject
+from direct.showbase.ShowBase import ShowBase
+from math import pi, sin, cos
+
+from utility.utils import exclude_from_dict
+from simulator.controllers.controller import Controller
+
+class CameraController(Controller, DirectObject):
+    __base: ShowBase
+
+    def __init__(self, *args, **kwargs):
+        Controller.__init__(self, *args, **exclude_from_dict(kwargs, ["camera", "origin"]))
+
+        self.__base = self._services.graphics.window
+        self.__cam = kwargs["camera"] if "camera" in kwargs else self.__base.cam
+        self.__origin = kwargs["origin"]
+
+        self.target_size = 6.37800000E+06
```

Comment:
im assuming all of the constants here will be stored as default values somewhere in a diff file or at the top of this file eventually

Reply:
as long as they're sane defaults, I think it'll do for now